File: 2.coding-level-one/6.randomness/cpp/1.basic simulation.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
using namespace std;

//simulation of coin toss experiment
void simulateCoinToss() {
	srand(time(0));
	cout << rand() % 2 << endl;
}

//simulation of dice throw experiment
void simulateDiceThrow() {
	srand(time(0));
	cout << rand() % 6 << endl;
}


/*int main() {
	simulateCoinToss();
	simulateDiceThrow();
}*/

File: 2.coding-level-one/6.randomness/java/1.BasicSimulation.java
import java.util.Random;

public class BasicSimulation {

	// simulation of coin toss
	public static void simulateCoinToss() {
		Random r = new Random();
		System.out.println(r.nextInt(2));
	}

	// simulation of dice throw experiment
	public static void simulateDiceThrow() {
		Random r = new Random();
		System.out.println(r.nextInt(6));
	}

	public static void main(String[] args) {
		simulateCoinToss();
		simulateDiceThrow();
	}
}
